app/services/middleware_topology_service.rb

```diff
@@ -52,11 +64,13 @@ def build_entity_data(entity)
   end
 
   def glyph?(entity)
-    [MiddlewareDatasource, MiddlewareDeployment].any? { |klass| entity.kind_of? klass }
+    [MiddlewareDatasource, MiddlewareDeployment, MiddlewareDomain, MiddlewareServerGroup]
+      .any? { |klass| entity.kind_of? klass }
   end
 
   def build_kinds
-    kinds = [:MiddlewareServer, :MiddlewareDeployment, :MiddlewareDatasource, :MiddlewareManager]
+    kinds = [:MiddlewareDeployment, :MiddlewareDatasource, :MiddlewareDomain, :MiddlewareManager,
+             :MiddlewareServer, :MiddlewareServerGroup]
```

Comment:
nit: when adding new things, worth to put them in the end, so it'll be easier to spot what was added.